### PR TITLE
Fix clippy warning

### PR DESCRIPTION
**Pull request by vitvakatu**
### Definition of Done

- [ ] There are no TODOs left in the code
- [ ] Change is covered by automated [tests](https://github.com/jni-rs/jni-rs/blob/master/CONTRIBUTING.md#tests)
- [ ] The [coding guidelines](https://github.com/jni-rs/jni-rs/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [ ] Public API has documentation
- [ ] This change is not breaking **or** mentioned in the Changelog
- [ ] The [continuous integration build](https://www.travis-ci.org/jni-rs/jni-rs) passes

### CODE REVIEW

File: src/wrapper/jnienv.rs [hunk 1]
```diff
@@ -1800,7 +1800,7 @@ impl<'a> JNIEnv<'a> {
             // attempt to acquire the lock. This prevents us from consuming the
             // mutex if there's an outstanding lock. No one else will be able to
             // get a new one as long as we're in the guarded scope.
-            let _ = mbox.try_lock()?;
+            std::mem::drop(mbox.try_lock()?);
```
Comment (dmitry-timofeev) [inline src/wrapper/jnienv.rs]: What is happening here? Why does it immediately unlock the acquired mutex? 😕

Comment (vitvakatu) [reply src/wrapper/jnienv.rs]: As we receiving this mutex via raw pointer, we're checking that the mutex isn't locked by anyone right now. Actually I'm not sure if it's a valid behaviour or not

Comment (dmitry-timofeev) [reply src/wrapper/jnienv.rs]: Ok, so it
1. Locks Java object (using its monitor)
2. Tries to lock the native object (using the mutex it is wrapped in)
3. If it is unlocked — clears the "Rust field" of the Java object and returns the unlocked mutex.
4. Finally , unlocks the Java monitor.

Comment (DarkEld3r) [inline src/wrapper/jnienv.rs]:
```suggestion
            drop(mbox.try_lock()?);
```
Isn't drop imported by the prelude?

Comment (dmitry-timofeev) [reply src/wrapper/jnienv.rs]: Would it make sense to clarify anything a little?

Comment (vitvakatu) [reply src/wrapper/jnienv.rs]: Well, it is exactly as you wrote it. I think additional clarification is not necessary, docs of used methods and this comment are enough to understand the purpose of the code.

Comment (dmitry-timofeev) [reply src/wrapper/jnienv.rs]: >  I think additional clarification is not necessary, docs of used methods and this comment are enough to understand the purpose of the code. 

I described what happens, but not _why_ it happens. If the Java lock is used, then how Rust lock can be 'locked'? Does it acquisition outlive the Java monitor acquisition?

**Comment by dmitry-timofeev** [comment]
OK, I think we'll improve this documentation some time later 🙃

**Comment by dmitry-timofeev** [comment]
Thanks!

**Pull request merged**